interview-questions.md

```diff
@@ -8,33 +8,35 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: Object-oriented programming lenguage is a more direct way to write code in a file or component, Ruby being one of them. Ruby's 'implicit return' programing lenguage is commonly known for by developers because it allows users to write their code without running 'return' within a function iteration. Unlike, the functional programing lenguages JavaScript, that requires its users to run 'return' every time and iterations is done within a function.
 
-Researched answer:
+Researched answer: OOP programing is built with organizing the software design based on classes and respective objects, whereas functional programming relies on functions. 
 
 1. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: Floats are decimal numbers, integer are whole numbers and both are diffent datatypes in Ruby
 
-Researched answer:
+Researched answer: It's important to known when doing arithmetics in Ruby, the result will always be an specific datatype. An integer is a number without a decimal point define as datatype. A float is a floating-point number, which means it is a number that has a decimal place and is define as another datatype.
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: It means that Ruby's syntax does not need to run 'return' in a function's iteration. 
 
-Researched answer:
+Researched answer: Ruby takes the final statement inside the method as a returned value without using the return keyword. In other words, the method will return the result of the final line before the return end keyword.
 
 1. What is a block in Ruby?
 
-Your answer:
+Your answer: A block is the code written to define a function in Ruby's syntax
 
-Researched answer:
+Researched answer: Anonymous codes that take and input from arguments and return a value and can be pass to a method call.
 
 1. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: In a Ruby hash you extract a value by using the method .delete on the hash and declaring the value.
 
-Researched answer:
+Researched answer: Calling the hash followed by a key name within brackets grabs the value associated with that key.
```

Comment:
Nice work in this section!  There are a few questions that were a little thinner, make sure to continue to extend your answers when possible.